src/index.js

```diff
@@ -163,6 +170,9 @@ const MicroModal = (() => {
   // Keep a reference to the opened modal
   let activeModal = null
 
+  // Keep an array of open modals so that esc key only closes top-most modal
+  let openModals = []
```

Comment:
The `openModals` array keeps track of all open modals.  As new modals are added, they are pushed to this array.  As they are closed, they are popped from the array.